config/urls.py

```diff
@@ -1,28 +1,42 @@
+from allauth.account.views import SignupView
 from django.conf import settings
 from django.conf.urls.static import static
 from django.contrib import admin
+from django.core.mail import send_mail
 from django.urls import include, path
 from django.views import defaults as default_views
 from django.views.generic import TemplateView
-
+from django.views.generic import TemplateView
+from django.shortcuts import render, redirect
+from allauth.account.adapter import get_adapter
+from allauth.account.utils import complete_signup
+from allauth.exceptions import ImmediateHttpResponse
 from events.users.forms import UserSignupForm
+from django.contrib import messages
+
 
+class HomePageView(SignupView):
```

Comment:
@karthicksakkaravarti 
This is causing problems, now when a user is logged in and they try to go to the home page, there get redirected to user profile